hedera-node/docs/design/services/smart-contract-service/udpate_hts_smart_contract.md

```diff
@@ -0,0 +1,86 @@
+# Update Hedera Token Service Smart Contract
+
+## Purpose
+
+In order to support updating the Token Custom Fees via smart contract calls, we need to expose a new function in the Hedera Token Service Smart Contract. This document will define the architecture and implementation of the `tokenFeeScheduleUpdate` function that will extend tha capabilities of the Hedera Token Service Smart Contract.
+
+## References
+
+[HIP-514](https://hips.hedera.com/hip/hip-514) - HIP that introduces token management via smart contracts.\
+[HIP-206](https://hips.hedera.com/hip/hip-206) - HIP that enables the precompiled system contracts for Hedera Token Service.
+
+## Goals
+
+- Expose `tokenFeeScheduleUpdate` as a new function in the Hedera Token Service Smart Contract.
+- Implement the needed HTS system contract classes to support the new function.
+
+## Non Goals
+
+- The implementation of the HAPI operation, as it is already an existing feature.
+
+## Architecture
+
+The architecture for token fee schedule update follows the existing framework defined for handling all calls to the HederaTokenService system contract in the modularization services and is described in more detail in the Implementation section below.
+
+## Implementation
+
+### New Struct Definitions
+
+| Struct Name          | Description                                                      |
+|----------------------|------------------------------------------------------------------|
+| `TokenFeeSchedule`   | A struct that contains the tokenId and fee schedule information. |
+
+We will add a new struct definition to the `IHederaTokenService` interface to support the new function.
+
+```solidity
```

Comment:
I wonder if it makes sense to add just the fee fields into the struct and the `tokenId` as a separate field into the `tokenFeeScheduleUpdate()` function?  This way the struct can be reused in the `IHederaTokenService.sol` file

Reply:
Which by the way does correspond to the way the HAPI call works ... even though the protobufs are (otherwise, and necessarily) different.